### PR TITLE
Fix/pro-locators

### PR DESCRIPTION
**Pull request by stdavis**

### CODE REVIEW

File: forklift/LocatorPallet.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    - The locators are published to arcgis server
+    - The locators need to be published from the `shipTo` folder in the `secrets.py` file and that folder
+        needs to be registered as a data store in arcgis server
+
+Creating the locators
+    - Make sure your `Dev` secrets are populated as that configuration will be used
+    - In arcgis pro python execute `LocatorsPallet.py`
+'''
+
+from glob import iglob
```
Comment (steveoh) [inline forklift/LocatorPallet.py:20]: **issue**: [path.glob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) makes this dependency redundant

Comment (steveoh) [inline forklift/LocatorPallet.py:20]: **chore**: remove dependency

File: forklift/LocatorPallet.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            "'AlternateStreetName.STREET_NAME AtlNamesAddrPnts.StreetName'",
+            "'AlternateStreetName.STREET_SUFFIX_TYPE AtlNamesAddrPnts.StreetType'",
+            "'AlternateStreetName.STREET_SUFFIX_DIR AtlNamesAddrPnts.SuffixDir'"
+        ]
+
+        start_seconds = perf_counter()
+        process_seconds = perf_counter()
+        self.log.info('creating the %s locator', 'address point')
+        try:
+            output_location = str(Path(self.output_location) / 'AddressPoints_AddressSystem')
```
Comment (steveoh) [inline forklift/LocatorPallet.py:218]: **suggestion**: save properties as Path classes on the instance e.g. (self.output_location, self.locators) and call `str` when necessary to remove the calls to the path constructor?

**Review by steveoh — COMMENTED** [review]
thanks scott, this is great!

File: forklift/LocatorPallet.py (hunk 1 above)
Comment (steveoh) [reply forklift/LocatorPallet.py:20]: sorry, but the path.glob comment didn't post. So with that refactor it'd be unnecessary.

File: forklift/LocatorPallet.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    - The locators are published to arcgis server
+    - The locators need to be published from the `shipTo` folder in the `secrets.py` file and that folder
+        needs to be registered as a data store in arcgis server
+
+Creating the locators
+    - Make sure your `Dev` secrets are populated as that configuration will be used
+    - In arcgis pro python execute `LocatorsPallet.py`
+'''
+
+from os.path import split
```
Comment (steveoh) [inline forklift/LocatorPallet.py:20]: **chore**: might as well use pathlib for this as well.

File: forklift/LocatorPallet.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+            'atlnamesroads': 'Roads_AddressSystem_STREET',
+            'geocoderoads': 'Roads_AddressSystem_STREET',
+        }
+
+        return set([lookup[crate.source_name.lower()] for crate in self.get_crates() if crate.was_updated()])
+
+    def copy_locator_to(self, file_path, locator, to_folder):
+        self.log.debug('copying %s to %s', file_path / locator, to_folder)
+        for filename in file_path.glob(f'{locator}.lo*'):
+            _, locator_with_extension = split(filename)
```
Comment (steveoh) [inline forklift/LocatorPallet.py:177]:
```suggestion
            locator_with_extension = filename.name
```

**Review by steveoh — COMMENTED** [review]
this is looking really close

**Comment by stdavis** [comment]
I messed up the rebase on this a bit so it copied some older commits. But my commits are a bit cleaner now.

**Review by steveoh — APPROVED** [review]

**Comment by steveoh** [comment]
This looks great. We need to chat about when this can be merged.

**Pull request merged**